### PR TITLE
[Merged by Bors] - feat: create carousel handler (VF-3748)

### PR DESCRIPTION
**Pull request by viniciusdacal**
**Fixes or implements VF-3748**

### Brief description. What is this change?
Added a new runtime handler to support carousel step.

With this particular step, we need to support a blocking flow, when the doesn't have a next port, and also a regular flow.
Each button inside the carousel, can trigger a command, and it should also have the possibility to add a no reply and no match.

### CODE REVIEW

File: lib/services/runtime/handlers/carousel.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+          title: replaceVariables(card.title, variablesMap),
+          description: {
+            slate,
+            text,
+          },
+          buttons: card.buttons.map((button) => ({
+            ...button,
+            name: replaceVariables(button.name, variablesMap),
+          })),
+        } as BaseNode.Carousel.TraceCarouselCard;
```
Comment (jonahsnider) [inline lib/services/runtime/handlers/carousel.ts:52]: why is this assertion necessary?

File: lib/services/runtime/handlers/carousel.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+import { StorageType } from '../types';
+import { slateInjectVariables, slateToPlaintext } from '../utils';
+import CommandHandler from './command';
+import { findEventMatcher } from './event';
+import NoMatchHandler from './noMatch';
+import NoReplyHandler, { addNoReplyTimeoutIfExists } from './noReply';
+
+const handlerUtils = {
+  _sample,
```
Comment (jonahsnider) [inline lib/services/runtime/handlers/carousel.ts:16]: this is unused

**Review by jonahsnider — APPROVED** [review]
looks good. needs some tests but this should be good to merge otherwise

File: lib/services/runtime/handlers/carousel.ts (hunk 1 above)
Comment (viniciusdacal) [reply lib/services/runtime/handlers/carousel.ts:52]: hmm, not needed actually, I added it to get cleaner error messages while implementing it.
Removing...

**Review by z4o4z — APPROVED** [review]

**Review by trs — APPROVED** [review]

**Pull request closed without merging**